### ISSUE TITLE
unexpected nested fillbox behaviour

### ISSUE THREAD

**Issue by kwadl** [DualSPHysics/DesignSPHysics#34]
When trying to fill an STL mesh to create a complex solid object floating in water, the fillbox used to generate the water (which intersects the fillbox used to fill the STL mesh) behaves unpredictably. Is this behaviour intended? The order of entities in my test case seems correct, and if the water fillbox is omitted the case builds as expected, just without water particles.

[fill_test.zip](https://github.com/DualSPHysics/DesignSPHysics/files/3957109/fill_test.zip)

**Comment by zlotny** [comment]
Hi there! First of all thanks for submitting the issue.

The result I get for your case (just open and generate gencase) is this:

![image](https://user-images.githubusercontent.com/6155245/70787568-3c743900-1d8f-11ea-9e8c-00e8ae49f712.png)

I assume this is correct. And I also assume your problem is due to an already fixed bug. I noticed your case was made on 0.6.0.1911-21-3, which is near of a month ago, and I was whacking a lot of bugs these last weeks :)

So in short, please try to update your DesignSPHysics version (from the Addon Manager) and see what happens with your case. You may have to change some MKs as some things changed around that a while ago but everything should just work ok.

Let me know if the screenshot was not the intended result or if you have any other problem. Thanks again!

**Comment by kwadl** [comment]
Yup, absolutely working now, must have been fixed in the last month :) 👍
